src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/Services/RazorSemanticTokensInfoService.cs

```diff
@@ -31,50 +26,44 @@ namespace Microsoft.AspNetCore.Razor.LanguageServer.Semantic;
 internal class RazorSemanticTokensInfoService(
     IRazorDocumentMappingService documentMappingService,
     RazorSemanticTokensLegendService razorSemanticTokensLegendService,
+    ICSharpSemanticTokensProvider csharpSemanticTokensProvider,
     LanguageServerFeatureOptions languageServerFeatureOptions,
-    IRazorLoggerFactory loggerFactory,
-    ITelemetryReporter? telemetryReporter)
+    IRazorLoggerFactory loggerFactory)
     : IRazorSemanticTokensInfoService
 {
     private const int TokenSize = 5;
 
     private readonly IRazorDocumentMappingService _documentMappingService = documentMappingService;
     private readonly RazorSemanticTokensLegendService _razorSemanticTokensLegendService = razorSemanticTokensLegendService;
+    private readonly ICSharpSemanticTokensProvider _csharpSemanticTokensProvider = csharpSemanticTokensProvider;
     private readonly LanguageServerFeatureOptions _languageServerFeatureOptions = languageServerFeatureOptions;
     private readonly ILogger _logger = loggerFactory.CreateLogger<RazorSemanticTokensInfoService>();
-    private readonly ITelemetryReporter? _telemetryReporter = telemetryReporter;
 
-    public async Task<SemanticTokens?> GetSemanticTokensAsync(
-        IClientConnection clientConnection,
-        TextDocumentIdentifier textDocumentIdentifier,
-        Range range,
+    public async Task<int[]?> GetSemanticTokensAsync(
         VersionedDocumentContext documentContext,
+        LinePositionSpan span,
         bool colorBackground,
+        Guid correlationId,
         CancellationToken cancellationToken)
     {
-        var correlationId = Guid.NewGuid();
-        using var _ = _telemetryReporter?.TrackLspRequest(Methods.TextDocumentSemanticTokensRangeName, LanguageServerConstants.RazorLanguageServerName, correlationId);
-
-        var semanticTokens = await GetSemanticTokensAsync(clientConnection, textDocumentIdentifier, range, documentContext, correlationId, colorBackground, cancellationToken).ConfigureAwait(false);
+        var semanticTokens = await GetSemanticTokensAsync(documentContext, span, correlationId, colorBackground, cancellationToken).ConfigureAwait(false);
 
-        var amount = semanticTokens is null ? "no" : (semanticTokens.Data.Length / TokenSize).ToString(Thread.CurrentThread.CurrentCulture);
+        var amount = semanticTokens is null ? "no" : (semanticTokens.Length / TokenSize).ToString(Thread.CurrentThread.CurrentCulture);
```

Comment:
>amount [](http://example.com/codeflow?start=12&length=6)

nit: this will run even if we are not logging at Info level. Doesn't seem that bad though.

Reply:
I'm not sure any of our logging is smart enough to not run, but everything in this PR is pre-existing so I'm going to leave it :)